### PR TITLE
Extract grammar into dedicated module

### PR DESCRIPTION
**Pull request by martint**
This allows us to keep the grammar on Java 8 for the CLI while being able to evolve the supporting parser infrastructure with Java 17 and newer version as the project updates.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: core/trino-grammar/pom.xml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    <parent>
+        <groupId>io.trino</groupId>
+        <artifactId>trino-root</artifactId>
+        <version>423-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>trino-grammar</artifactId>
+    <name>trino-grammar</name>
```
Comment (electrum) [inline core/trino-grammar/pom.xml:13]: We don't use this element anymore

**Review by electrum — APPROVED** [review]

File: client/trino-cli/pom.xml [hunk 2]
```diff
@@ -139,6 +151,24 @@
...
+                        <artifactId>surefire-junit-platform</artifactId>
+                        <version>${dep.plugin.surefire.version}</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>org.apache.maven.surefire</groupId>
+                        <artifactId>surefire-testng</artifactId>
+                        <version>${dep.plugin.surefire.version}</version>
+                    </dependency>
+                </dependencies>
+            </plugin>
```
Comment (ksobolew) [inline client/trino-cli/pom.xml:170]: Doesn't seem like it belongs here. I don't see any changes in tests?

File: core/trino-parser/pom.xml [hunk 3]
```diff
@@ -13,7 +13,6 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <project.build.targetJdk>8</project.build.targetJdk>
```
Comment (ksobolew) [inline core/trino-parser/pom.xml:16]: Could we do it as a separate commit, kind of like a cherry-on-top? :)

File: client/trino-cli/pom.xml (hunk 2 above)
Comment (martint) [reply client/trino-cli/pom.xml:170]: Some classes moved from the old parser module to the CLI module and they rely on junit for tests

File: core/trino-parser/pom.xml (hunk 3 above)
Comment (martint) [reply core/trino-parser/pom.xml:16]: What’s the benefit? It’s part of the refactoring — the parts that depend on Java 8 moved to the grammar module, so that constraint moved there, too, at the same time.

**Review by dain — APPROVED** [review]
